src/apps/vpn/models/serverdata.cpp

```diff
@@ -208,6 +209,51 @@ void ServerData::changeServer(const QString& countryCode,
     return;
   }
 
+  if (entryCountryCode.isEmpty() && entryCityName.isEmpty()) {
+    logger.info() << "Single hop server change";
+    mozilla::glean::sample::user_changed_endpoint_geo.record(
+        mozilla::glean::sample::UserChangedEndpointGeoExtra{
+            ._server = "SingleHop",
+        });
+  } else {
+    logger.info() << "Multihop server change";
+    // If exit changed, or changing from single hop to multihop
+    // (could have same exit for single and multihop, and we'd want to record
+    // metric)
+    if (m_exitCountryCode != countryCode || m_exitCityName != cityName ||
+        (m_entryCountryCode.isEmpty() && m_entryCityName.isEmpty())) {
+      logger.info() << "Multihop exit server change";
+      mozilla::glean::sample::user_changed_endpoint_geo.record(
+          mozilla::glean::sample::UserChangedEndpointGeoExtra{
+              ._server = "MultiHopExit",
+          });
+    }
+
+    // Cannot use an if/else, as it's possible both exit and entry changed
+    if (m_entryCountryCode != entryCountryCode ||
+        m_entryCityName != entryCityName) {
+      logger.info() << "Multihop entry server change";
+      mozilla::glean::sample::user_changed_endpoint_geo.record(
+          mozilla::glean::sample::UserChangedEndpointGeoExtra{
+              ._server = "MultiHopEntry",
+          });
+    }
+  }
+
+  update(countryCode, cityName, entryCountryCode, entryCityName);
+}
+
+void ServerData::changeServerForTutorial(const QString& countryCode,
```

Comment:
This feels like an unnecessary change given tutorials as a feature are about to go away.

Reply:
Without this, we're going to count tutorial runs as server changes, which is not what we want (I think). I'd rather include this code than risk counting this wrong in 2.17 (if Super Dooper is live in 2.17, but for some reason tutorials are not removed until 2.18).